### DISCUSSION TITLE
Alert export limited to organization admins

### DISCUSSION THREAD

**Discussion by mahradk** [grafana/grafana#66394] [Alerting]
Hey,

I was wondering if it is possible to allow export of alerts with Editor/Viewer role as this is an standard feature of the dashboards.

As the [provisioning API](https://grafana.com/docs/grafana/latest/developers/http_api/alerting_provisioning/#span-idroute-get-alert-rule-exportspan-export-an-alert-rule-in-provisioning-file-format-_routegetalertruleexport_) is the only way one could get to unified alert rule JSONs.
If an `Editor` that has created an alert wants to export their alerts for provisioning, they have to contact an `Org. Admin`.

Would it be possible to update `alertingReaderRole` and `alertingWriterRole` of [ngalert/accesscontrol](https://github.com/grafana/grafana/blob/main/pkg/services/ngalert/accesscontrol.go#L135-L174) to allow `ActionAlertingProvisioningRead` and `ActionAlertingProvisioningWrite` respectively?

**Comment by soniaAguilarPeiron** [comment]
This export is only available if user has the necessary [permissions ](https://github.com/grafana/grafana/blob/559da8b3474dee6d966672b7b2cb7b7fa057c8cc/pkg/services/ngalert/api/authorization.go#L195-L209)=> `alert.provisioning:read` **with RBAC** ,falling back to `admin` **without RBAC**.
You can use this `alert.provisioning:read`, or is not possible in your case?

**Reply by cmergenthaler** [reply]
Would be nice to allow the export of alert rules to `Editor's`, as they are also allowed to create the alert rules even without RBAC.

**Reply by mahradk** [reply]
@soniaAguilarPeiron Thanks for the response.
You guessed it, it is not possible in this case.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82735. Feel free to continue the discussion around this there. Thank you!
